src/vos/vos_io.c

```diff
@@ -2113,6 +2162,12 @@ vos_update_begin(daos_handle_t coh, daos_unit_oid_t oid, daos_epoch_t epoch,
 		", flags="DF_X64"\n", DP_UOID(oid), iod_nr,
 		dtx_is_valid_handle(dth) ? dth->dth_epoch :  epoch, flags);
 
+	rc = vos_check_akeys(iod_nr, iods);
```

Comment:
I need to debug more what the real problem is here but it was disallowed by the timestamp cache before.   I realize it's n^2 here but my expectation is that n is usually small.

Without this, I get valgrind errors in rollback of aborted transactions we have some other assumption somewhere that causes issues.

Reply:
I mean that when we do the real update, if we probe an akey that is just one of inserted by current update, then it will be non-committed status, we will find that such akey's DTX owner is just current DTX. Then we can know it is a duplicated akey. I am not sure whether it can resolve your issue.

Reply:
I don't see this as a big issue.  Yes it's n^2 but I think practically, we are talking about a handful of akeys at most in a single vos operation, not thousands.

I'm sure it's possible to do what you suggest but I'd rather leave that for future work because it's somewhat tricky.   We currently treat things written by the same transaction as committed.  I really need to rework some of what we do in VOS to simplify the logic.

Reply:
I agree that for most of cases, the akey number in single VOS operation will be very limited, we can consider the optimization that after the demo.